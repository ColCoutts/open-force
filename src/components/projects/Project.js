import React from 'react';
import PropTypes from 'prop-types';
//bring in Link to get to Project Detail page (Link to={project.id})

function Project({ project }) {
  return (
    <section>
      <h1>{project.title}</h1>
      <a href={project.url}>{project.url}</a>
      <p>{project.summary}</p>
    </section>
  );
}

Project.propTypes = {
  project: PropTypes.shape({
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
    imgUrl: PropTypes.string.isRequired,
    summary: PropTypes.string.isRequired,
    id: PropTypes.string.isRequired
  }).isRequired
};

export default Project;
