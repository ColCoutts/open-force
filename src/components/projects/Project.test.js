import React from 'react';
import { shallow } from 'enzyme';
import Project from './Project';

describe('project component', () => {
  it('renders a snapshot for project', () => {
    const project = {
      title: 'Project',
      url: './url',
      imgUrl: './image',
      summary: 'text',
      id: 'someIdthing'
    };
    const wrapper = shallow(<Project project={project} />);
    expect(wrapper).toMatchSnapshot();
  });
});
