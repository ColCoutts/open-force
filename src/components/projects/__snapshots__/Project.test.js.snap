// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`project component renders a snapshot for project 1`] = `
<section>
  <h1>
    Project
  </h1>
  <a
    href="./url"
  >
    ./url
  </a>
  <p>
    text
  </p>
</section>
`;
